replaced all instances of Block to Bundle

dkuthoore opened the pull request:
## Summary
Refactors the Block tracking contract and related tests/docs to use "Bundle" semantics, aligning naming with protocol direction. Ethereum built-ins like `block.timestamp` remain unchanged. No vendored or generated files were modified.

## Changes

### Contracts
Updated `src/implementation/BlockTracker.sol` in-place:
- Contract: `BlockTracker` → `BundleTracker`
- Events:
  - `BlockProposed` → `BundleProposed`
  - `BlockTrackerDeployed` → `BundleTrackerDeployed`
- Public mapping:
  - `blocks` → `bundles`
- Function:
  - `proposeBlock(string _blockData)` → `proposeBundle(string _bundleData)`
- Parameters/locals/comments:
  - `blockData` → `bundleData`
  - `blockProposer` → `bundleProposer`
  - All docstrings and inline comments: "block" → "bundle" where not an Ethereum built-in
- Preserved Ethereum built-ins: `block.timestamp` unchanged
- changed name of file

### Tests
Updated `test/BlockTracker.t.sol` in-place:
- Contract: `BlockTrackerTest` → `BundleTrackerTest`
- Instance: `BlockTracker blockTracker` → `BundleTracker bundleTracker`
- Address: `blockProposer` → `bundleProposer`
- Test: `testProposeBlock()` → `testProposeBundle()`
- Calls and getters:
  - `proposeBlock(...)` → `proposeBundle(...)`
  - `blocks(...)` → `bundles(...)`
- Updated all inherited function calls to reference `bundleTracker`
- changed name of file

### Documentation
Updated root `README.md`:
- Renamed section and references:
  - "BlockTracker" → "BundleTracker"
  - "Block Proposal / blocks / BlockProposed / BlockTrackerDeployed" → "Bundle Proposal / bundles / BundleProposed / BundleTrackerDeployed"
- Deployment example target updated to `...:BundleTracker`
- Intro line updated from "Blocks of transactions..." to "Bundles of transactions..."
- Kept Ethereum built-ins unchanged; docs directory not edited

## Exclusions
- Did not change any Ethereum built-ins or EVM semantics (`block.timestamp`, `block.number`, `blockhash`).
- Did not modify generated artifacts under `out/**` or docs under `docs/**`.
- Did not touch vendored dependencies under `lib/**`.

## Testing
- Ran forge tests.
- All tests passed:

KagemniKarimu commented:
i was not able to very competently review the solidity , but the other changes looked good 👍🏿

dkuthoore commented:
@pemulis: per @KagemniKarimu 's comment above ^
reviewing on GH makes it seem like i rewrote the entire solidity contract, but i only made edits to the existing "BlockTracker.sol" contract (and the corresponding test contract). after the edits, i changed the name of the file, thus it shows up on here like all lines are newly added. 

so i wouldn't spend too much time reviewing the logic of contract, but just the instances of 'Bundle' (or looking for any instances of 'Block', there shouldnt be any besides 'block.timestamp')

also:
<img width="800" height="493" alt="image" src="https://github.com/user-attachments/assets/22a93039-9499-42c2-ba86-356620930c55" />
